plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergGlueCatalogAccessOperations.java

```diff
@@ -0,0 +1,502 @@
+import static io.trino.plugin.iceberg.GlueMetastoreMethod.CREATE_TABLE;
+import static io.trino.plugin.iceberg.GlueMetastoreMethod.GET_DATABASE;
+import static io.trino.plugin.iceberg.GlueMetastoreMethod.GET_TABLE;
+import static io.trino.plugin.iceberg.TableType.DATA;
+import static io.trino.plugin.iceberg.TableType.FILES;
+import static io.trino.plugin.iceberg.TableType.HISTORY;
+import static io.trino.plugin.iceberg.TableType.MANIFESTS;
+import static io.trino.plugin.iceberg.TableType.PARTITIONS;
+import static io.trino.plugin.iceberg.TableType.PROPERTIES;
+import static io.trino.plugin.iceberg.TableType.SNAPSHOTS;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.fail;
+
+/*
+ * TestIcebergGlueCatalogAccessOperations currently uses AWS Default Credential Provider Chain,
+ * See https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
+ * on ways to set your AWS credentials which will be needed to run this test.
+ */
+@Test(singleThreaded = true)
+public class TestIcebergGlueCatalogAccessOperations
+        extends AbstractTestQueryFramework
+{
+    private final String schema = "test_schema_" + randomTableSuffix();
+    private final Session TEST_SESSION = testSessionBuilder()
+            .setCatalog("iceberg")
+            .setSchema(schema)
+            .build();
+
+    private GlueMetastoreStats catalogStats;
+    private GlueMetastoreStats tableOperationStats;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        File tmp = Files.createTempDirectory("test_iceberg").toFile();
+        DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(TEST_SESSION).build();
+
+        AtomicReference<GlueMetastoreStats> catalogStatsReference = new AtomicReference<>();
+        AtomicReference<GlueMetastoreStats> tableOperationStatsReference = new AtomicReference<>();
+
+        queryRunner.installPlugin(new TestingIcebergPlugin(
+                Optional.empty(),
+                Optional.empty(),
+                Optional.of(new StealStatsModule(catalogStatsReference, tableOperationStatsReference))));
+        queryRunner.createCatalog("iceberg", "iceberg",
+                ImmutableMap.of(
+                        "iceberg.catalog.type", "glue",
+                        "hive.metastore.glue.default-warehouse-dir", tmp.getAbsolutePath()));
+
+        queryRunner.execute("CREATE SCHEMA " + schema);
```

Comment:
in case of test cancellation, will this get cleaned on AWS by something?

cc @alexjo2144 @jirassimok @homar

Reply:
I am not aware of any mechanism that would clear this

Reply:
@findepi I've seen "register autoclosable" around. Is it worth trying to figure out a way to register in this schema to auto close? Not sure that would solve the problem here vs behavior change vs the `@Afterclass` function.

Reply:
the problem i am thinking about is abrupt JVM termination that breaks tests execution. Like CI node cancellation or failure. 
we have https://github.com/trinodb/trino/blob/7eb9c573b75e08c583031c754edef591ab82ee55/plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastore.java#L227-L260
but it won't cover here because the name doesn't match.
Maybe we just extend the name pattern to nuke everything starting with "test_" there.